sys/registry/registry.c

```diff
@@ -0,0 +1,198 @@
+#include <stdio.h>
+#include <string.h>
+#include <errno.h>
+#include "clist.h"
+#include "registry/registry.h"
+#include "kernel_defines.h"
+#include "assert.h"
+#define ENABLE_DEBUG (0)
+#include "debug.h"
+
+static int _registry_cmp_name(clist_node_t *current, void *name);
+static int _registry_call_commit(clist_node_t *current, void *name);
+static void _parse_name(char *name, int *name_argc, char *name_argv[]);
+static registry_handler_t *_handler_parse_and_lookup(char *name, int *name_argc,
+                                                     char *name_argv[]);
```

Comment:
These are C prototypes. If not present, the compiler might not be able to resolve a function if it wasn't defined before

Reply:
yah sure, all of these have definitions higher in the code than their first use. so the prototypes aren't needed. It compiles if you comment out all of these

Reply:
I don't think there's something wrong with having the prototypes here. In case the functions are touched, there might be some cases where a function is not reached. Also, it doesn't increase binary size nor readability.